### PR TITLE
Remove extra tab stops in Reader Document View

### LINKED ISSUE #12245: Reader Document Viewer: Page loses visual indication of focus after tabbing through `Search cases`
When a document is selected to be displayed in the reader, there is no visual indication of focus after the Search Text element. (New: 08/27/2019) I used my tool to show the focus in yellow below however you cannot visually see this focus without the tool on.  This loss of focus happens for seven tab stops.

Screen Shot 2019-09-27 at 5 18 59 PM

### PR DESCRIPTION
**Pull request by yoomlam**
Resolves #12245

### Description
Remove extra tab stops:
1. for search bar when it is hidden and
2. for the PDF view, i.e.,`ReactVirtualized__Grid` elements (`Grid` React components within `PdfFile`)

See https://github.com/department-of-veterans-affairs/caseflow/issues/12245#issuecomment-644406955

### Acceptance Criteria
- [x] Code compiles correctly
- [ ] Tab stops flow from search icon to the PDF to the bottom of PDF viewer when search bar is hidden
- [ ] Tab stops flow from search icon to search bar elements when search bar is visible
- [ ] Reader's keyboard shortcuts still work

### Testing Plan
1. Log in as attorney
2. Go to their queue and click one of the "View docs" links for an appeal
3. Use keyboard to tab to the search (magnifying glass) icon. Continue to press Tab to see the next selection/focus.
   * Try clicking on the search icon to open the search panel; then pressing Tab.
   * Try Shift-Tab to go backwards.

All Tab keyboard navigation should move the current focus to a reasonable element.

### User Facing Changes

The current focused element should always be visible with a dotted outline.

### CODE REVIEW

**Comment by yoomlam** [comment]
Keyboard conflict with 'space'.
When using the keyboard, space is usually used to activate the focused element.
This conflicts with Reader's shortcut for scrolling down:
image

Easiest fix is to change Reader's shortcut, but how will this affect current users who use space to scroll down?

File: client/app/reader/PdfFile.jsx [hunk 1]
```diff
@@ -469,6 +470,7 @@ export class PdfFile extends React.PureComponent {
             columnWidth={this.getColumnWidth}
             columnCount={this.columnCount}
             scale={this.props.scale}
+            tabIndex={this.props.isVisible ? 0 : -1}
```
Comment (yoomlam) [inline client/app/reader/PdfFile.jsx:473]: Skip tab focus (`-1`) if element is not visible.

File: client/app/reader/PdfFile.jsx [hunk 2]
```diff
@@ -480,11 +482,41 @@ export class PdfFile extends React.PureComponent {
...
+  setDocScrollPosition: PropTypes.func,
+  setPageDimensions: PropTypes.func,
+  setPdfDocument: PropTypes.func,
+  showPlaceAnnotationIcon: PropTypes.func,
+  sidebarHidden: PropTypes.bool,
+  startPlacingAnnotation: PropTypes.func,
+  togglePdfSidebar: PropTypes.func,
+  updateSearchIndexPage: PropTypes.func,
+  updateSearchRelativeIndex: PropTypes.func,
+  windowingOverscan: PropTypes.number
```
Comment (yoomlam) [inline client/app/reader/PdfFile.jsx:519]: Appeasing the linter gods!

File: client/app/styles/react/_search_bar.scss [hunk 3]
```diff
@@ -99,6 +99,7 @@
 
   &.hidden {
     top: 0;
+    display: none;
```
Comment (yoomlam) [inline client/app/styles/react/_search_bar.scss:102]: This was the suggested solution mentioned [here](https://stackoverflow.com/questions/42761987/tabindex-1-not-working-for-child-elements).

File: spec/feature/reader/reader_spec.rb [hunk 4]
```diff
@@ -1232,10 +1232,10 @@ def open_search_bar
 
     scenario "Show and Hide Document Searchbar with Keyboard" do
       visit "/reader/appeal/#{appeal.vacols_id}/documents/#{documents[0].id}"
-      search_bar = find(".cf-search-bar")
 
       find("body").send_keys [:meta, "f"]
 
+      search_bar = find(".cf-search-bar")
```
Comment (yoomlam) [inline spec/feature/reader/reader_spec.rb:1238]: Can't find it until it is displayed.

**Comment by geronimoramos** [comment]
> Keyboard conflict with 'space'.
> When using the keyboard, space is usually used to activate the focused element.
> This conflicts with Reader's shortcut for scrolling down:
> image
> 
> Easiest fix is to change Reader's shortcut, but how will this affect current users who use space to scroll down?

I played with the tabbing on this screen. It looks like users are still able to use space to scroll down when they are focused on the document element. And they are still able to use space to interact with other elements when they're focused on it. Looks operational to me! @yoomlam

**Comment by yoomlam** [comment]
> I played with the tabbing on this screen. It looks like users are still able to use space to scroll down when they are focused on the document element. And they are still able to use space to interact with other elements when they're focused on it. Looks operational to me! @yoomlam

To clarify, the UX problem occurs when the user can use any Reader keyboard shortcuts **except 'space'** when focused on other elements.

**Pull request merged**